src/lib/storage/table.cpp

```diff
@@ -59,6 +59,25 @@ Table::Table(const TableColumnDefinitions& column_definitions, const TableType t
   }
 }
 
+void Table::delete_column(ColumnID column_id) {
```

Comment:
The changes required to other classes for the realization of the ALTER TABLE DROP COLUM functionality should be as small as possible. Similar to `delete_segment(...)`, `delete_column(...)` should not implicitly remove additional data structures other than the actual column of the table. Removing the column and removing the indexes created for the column is the responsibility of the ALTER TABLE operator. Implementing `Table::remove_index` would complement `Table::create_index` and can then be called together with `delete_column` from the ALTER TABLE operator.